### DISCUSSION TITLE
Transparent Background For Extension With A Second Canvas

### DISCUSSION THREAD

**Discussion by PURPL3-CAT** [TurboWarp/extensions#751] [Q&A]
I'm currently making an extension that adds BabylonJS to TurboWarp. It works by adding another canvas behind the stage and rendering to that. The background is opaque, and I have to use a custom build of scratch-render. My extension works fine on my local build, but I want to move to using regular TurboWarp (especially if I share it, which I plan to do if I can). If anyone knows a solution (push to the actual scratch-render repo, overwrite the render code from the extension, etc..), that would be very helpful!

![Screenshot Capture - 2023-07-16 - 17-39-27](https://github.com/TurboWarp/extensions/assets/100864922/8d67d57e-659b-4a31-9a3d-bb01c138cafb)

**Comment by yuri-kiss** [comment]
yo this looks super cool.

**Reply by PURPL3-CAT** [reply]
Thanks!

**Comment by BlueDome77** [comment]
There is a block in Canvas Effects that allows you to set the transparency of the stage. (**Only** the stage, sadly) Link to the PR: #747

**Reply by PURPL3-CAT** [reply]
Ok thats too bad. Thanks anyway!

**Comment by PURPL3-CAT** [comment]
Update: the only way is if the WebGL context is created with alpha: true. I'll look for other solutions such as rendering the second canvas as a sprite on the main canvas.
